game/main.cc

```diff
@@ -130,7 +132,15 @@ void mainLoop(std::string const& songlist) {
 	Audio audio;
 	std::clog << "core/info: Loading assets." << std::endl;
 	TranslationEngine localization(PACKAGE);
-	Window window;
+	boost::optional<Window> window;
+	try {
+		window = boost::in_place();
+		} catch (RUNTIME_ERROR& e) {
+			std::cerr << "ERROR: " << e.what() << std::endl;
+		}
+		
+	if (!window) throw std::runtime_error("Unable to create window.");
+
 	TextureLoader m_loader;
```

Comment:
If window creation fails, you print a message and re-throw? This looks like something that could well be done within the Window constructor itself.